lib/Backend/PreLowerPeeps.cpp

```diff
@@ -13,6 +13,13 @@ IR::Instr *Lowerer::PreLowerPeepInstr(IR::Instr *instr, IR::Instr **pInstrPrev)
 
     switch (instr->m_opcode)
     {
+#if defined(_M_IX86) || defined(_M_X64)
+        // this sort of addressing mode magic only applies to x86 platforms
```

Comment:
It seems like our ARM encoder would not support it without a lot of work, and since this is mostly for asm.js/wasm where we don't have arm support i'll open an issue and leave this for now.